server/src/main/java/org/opensearch/index/mapper/BinaryFieldMapper.java

```diff
@@ -242,28 +242,26 @@ protected String contentType() {
      */
     public static class CustomBinaryDocValuesField extends CustomDocValuesField {
 
-        private final ArrayList<byte[]> bytesList;
+        private final TreeSet<byte[]> bytesSet;
```

Comment:
@kkmr TreeSet is a great structure but:
 - it consumes noticeably more memory
 - we also do not need to keep the values sorted all the time but only when binaryValue() is called (so we pay sort + dedup price only once)

Could you please estimate what the overhead this new structure would have? Thank you.

Reply:
+1 to run benchmarks. We never followed up on that the last time we poked this code. I don't think the modification in place is problematic and it keeps memory managed but let's verify this in a microbenchmark. 

Original conversation [here](https://github.com/opensearch-project/OpenSearch/pull/6884#discussion_r1153244732).